lint/rule_panel_rate_interval.go

```diff
@@ -14,24 +14,21 @@ func NewPanelRateIntervalRule() *PanelRuleFunc {
 		name:        "panel-rate-interval-rule",
 		description: "Checks that each panel uses $__rate_interval.",
 		fn: func(d Dashboard, p Panel) Result {
-			if t := getTemplateDatasource(d); t == nil || t.Query != "prometheus" {
+			if t := getTemplateDatasource(d); t == nil || t.Query != "prometheus" || !panelHasQueries(p) {
 				// Missing template datasources is a separate rule.
 				return Result{
 					Severity: Success,
 					Message:  "OK",
 				}
 			}
```

Comment:
In this situation a more verbose check would be preferred, so future editors don't miss it.

```suggestion
			if t := getTemplateDatasource(d); t == nil || t.Query != "prometheus" {
				// Missing template datasources is a separate rule.
				return Result{
					Severity: Success,
					Message:  "OK",
				}
			}
			
			if !panelHasQueries(p) {
				// Don't lint certain types of panels.
				return Result{
					Severity: Success,
					Message:  "OK",
				}
			}
```